src/components/Contentful/JourneyCardCarousel.vue

```diff
@@ -354,12 +341,39 @@ const handleChange = interaction => {
 	);
 };
 
-// Watch orderedSlides to update isLoading
-watch(orderedSlides, (newSlides, oldSlides) => {
-	if (oldSlides && JSON.stringify(oldSlides) !== JSON.stringify(newSlides)) {
-		isLoading.value = false;
+onMounted(() => {
+	try {
+		const contentfulChallengeResult = apollo.readQuery({
+			query: contentfulEntriesQuery,
+			variables: { contentType: 'challenge', limit: 200 }
+		});
+
+		const achievementsResult = apollo.readQuery({
+			query: userAchievementProgressQuery
+		});
+
+		const slidesResult = apollo.readQuery({
+			query: contentfulEntriesQuery,
+			variables: {
+				contentType: 'carousel',
+				contentKey: CONTENTFUL_CAROUSEL_KEY,
+			}
+		});
+
+		slides.value = slidesResult.contentful?.entries?.items?.[0]?.fields?.slides ?? [];
+
+		const badgeContentfulData = (contentfulChallengeResult.contentful?.entries?.items ?? [])
+			.map(entry => getContentfulLevelData(entry));
+
+		const badgeAchievementData = [
+			...(achievementsResult.userAchievementProgress?.tieredLendingAchievements ?? [])
+		];
+
+		badgesData.value = combineBadgeData(badgeAchievementData, badgeContentfulData);
+	} catch (e) {
+		logReadQueryError(e, 'MyKivaPage uiConfigSettingQuery');
```

Comment:
`uiConfigSettingQuery` isn't used here so this could be more generic like `MyKivaPage onMounted`